services/elasticsearch.go

```diff
@@ -972,22 +985,56 @@ func QueryElastic(
 	return results, nil
 }
 
-func GetElasticClient() (*opensearch.Client, error) {
+func GetElasticClientByType(instance_type string) (*opensearch.Client, error) {
 	mu.Lock()
 	defer mu.Unlock()
+	if instance_type == "primary" {
```

Comment:
Would it make sense to use an enum here? The compiler can then detect typos and it is clearer there can only be so many types. This also means it is impossible to hit that invalid condition at the end.

You can create an enum with an iota integer type